### ISSUE TITLE
Better control of the metrics reported by jenkins

### ISSUE THREAD

**Issue by stmlange** [jenkinsci/opentelemetry-plugin#690]
### What feature do you want to see added?

Hello,
thanks for the plugin!
We now started to use it to collect metrics as well and I must admit that our jenkins setup is rather complicated. In combination with docker the reporting of metrics is not really helpful at all since the docker agents just have a random number as their hostname.
Considering that we plan that the underlying application should report metrics too I think we need a better control and grip how the OTEL Plugin is reporting metrics. In general the application metrics should be persistent for a year, while I think the jenkins metrics is only useful to be stored for two weeks (or some other limit) or not created at all.
I'm aware that prometheus has flags like "--storage.tsdb.retention.time=365d", but this applies for all the (metric) data.

Hence I would like to have the ability to
1) either disable the metric reporting entirely
2) have the ability to somehow remove the metrics that are collected by jenkins after a specified time (e.g. 2 weeks) that is unrelated to the retention that can be configured in prometheus.

Let me know if this makes sense, I haven't found any options to achieve the above. We really would like to have multiple systems.

**Comment by cyrille-leclerc** [comment]
> reporting of metrics is not really helpful at all since the docker agents just have a random number as their hostname.

I'm not sure to follow:
- we define `service.instance.id` based on the Jenkins home unique identifier, it wouldn't change if you reuse your Jenkins home on a different host name
- can't you aggregate across the hostname?

> disable metrics exporter

Setting `otel.metrics.exporter=none` should do the job.

> 2 weeks metrics retention

This has to be configured in your metrics store, what do you use?

**Comment by stmlange** [comment]
Hello, thanks for your feedback!
I tried `otel.metrics.exporter=none` so my config now looks like:
```
otel.instrumentation.jenkins.web.enabled=false
otel.metrics.exporter=none
```
but that results in an unhandled exception:
```
Caught unhandled exception with ID ....................................................
io.opentelemetry.sdk.autoconfigure.spi.ConfigurationException: otel.metrics.exporter contains none along with other exporters
	at io.opentelemetry.sdk.autoconfigure.MeterProviderConfiguration.configureMetricReaders(MeterProviderConfiguration.java:64)
	at io.opentelemetry.sdk.autoconfigure.MeterProviderConfiguration.configureMeterProvider(MeterProviderConfiguration.java:51)
	at io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdkBuilder.build(AutoConfiguredOpenTelemetrySdkBuilder.java:348)
	at io.jenkins.plugins.opentelemetry.OpenTelemetrySdkProvider.initializeOtlp(OpenTelemetrySdkProvider.java:160)
	at io.jenkins.plugins.opentelemetry.OpenTelemetrySdkProvider.initialize(OpenTelemetrySdkProvider.java:131)
	at io.jenkins.plugins.opentelemetry.JenkinsOpenTelemetryPluginConfiguration.initializeOpenTelemetry(JenkinsOpenTelemetryPluginConfiguration.java:222)
	at io.jenkins.plugins.opentelemetry.JenkinsOpenTelemetryPluginConfiguration.configure(JenkinsOpenTelemetryPluginConfiguration.java:165)
	at jenkins.model.Jenkins.configureDescriptor(Jenkins.java:4039)
	at jenkins.model.Jenkins.doConfigSubmit(Jenkins.java:4003)
```

Does this perhaps require a specific version of jenkins / otel-plugin?

**Comment by cyrille-leclerc** [comment]
See
* https://github.com/jenkinsci/opentelemetry-plugin/issues/691
* https://github.com/jenkinsci/opentelemetry-plugin/pull/692

**Comment by cyrille-leclerc** [comment]
Fixed in 2.16.0
